Build broke on termux aarch64 clang-16

**Describe the bug**
Build broken on Termux/aarch64 (clang compiler)

**To Reproduce**
Steps to reproduce the behavior:
on aarch64 device running Termux
git pull up to 937cadf054c1cf9aa4cd852de39bb64c9ee49c7e
run cmake in main dir
run make -j8 in main dir

Observe:

```
 [  7%] Building C object src/cdrom/CMakeFiles/cdrom.dir/cdrom_image_backend.c.o
/data/data/com.termux/files/home/86Box/src/disk/hdc_st506_xt.c:627:13: error: expected expression
            [[fallthrough]];
            ^
/data/data/com.termux/files/home/86Box/src/disk/hdc_st506_xt.c:1612:13: error: expected expression
            [[fallthrough]];
            ^
2 errors generated.

```

and few more downstream

**Expected behavior
Build finishes

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - N/A

**Additional context**
clan-16 may be picky

Sadly, still fails:

```
~/86Box $ make -j 8                                                                                [  0%] Building C object src/CMakeFiles/vnc.dir/vnc.c.o
[  0%] Building C object src/cdrom/CMakeFiles/cdrom.dir/cdrom.c.o
[  0%] Building C object src/unix/CMakeFiles/ui.dir/unix_sdl.c.o
[  0%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_486.c.o
[  0%] Building C object src/chipset/CMakeFiles/chipset.dir/82c100.c.o
[  0%] Building C object src/cpu/CMakeFiles/cpu.dir/cpu.c.o                                        [  0%] Building C object src/cpu/808x/CMakeFiles/808x.dir/queue.c.o
[  0%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/f2xm1.cc.o
/data/data/com.termux/files/home/86Box/src/cpu/808x/queue.c:79:68: warning: format specifies type 'int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
        fatal("Requested prefetch queue of %i bytes is too big\n", size);
                                           ~~                      ^~~~
                                           %zu
/data/data/com.termux/files/home/86Box/src/cpu/cpu.c:1612:56: warning: format specifies type 'unsigned long long' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]
            fatal("cpu_set : unknown CPU type %llu\n", cpu_s->cpu_type);
                                              ~~~~     ^~~~~~~~~~~~~~~
                                              %lu
1 warning generated.
[  0%] Built target 808x
[  0%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_686.c.o
[  0%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/fpatan.cc.o
[  0%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/fprem.cc.o
[  1%] Building C object src/chipset/CMakeFiles/chipset.dir/acc2168.c.o
[  1%] Building C object src/CMakeFiles/vnc.dir/vnc_keymap.c.o
[  1%] Built target vnc
[  1%] Building C object src/chipset/CMakeFiles/chipset.dir/cs8230.c.o
[  1%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1429.c.o
[  1%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1435.c.o                            [  2%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/fsincos.cc.o
[  2%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_common.c.o
[  2%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1489.c.o                            [  2%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_k6.c.o
1 warning generated.
[  2%] Building C object src/cdrom/CMakeFiles/cdrom.dir/cdrom_image_backend.c.o
[  2%] Building C object src/cpu/CMakeFiles/cpu.dir/cpu_table.c.o
[  2%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen.c.o                        [  2%] Building C object src/device/CMakeFiles/dev.dir/bugger.c.o
/data/data/com.termux/files/home/86Box/src/unix/unix_sdl.c:159:26: warning: incompatible pointer types passing 'uint8_t[17842176]' (aka 'unsigned char[17842176]') to parameter of type 'uint32_t *' (aka 'unsigned int *') [-Wincompatible-pointer-types]                                                       video_screenshot(interpixels, 0, 0, 2048);
                         ^~~~~~~~~~~
/data/data/com.termux/files/home/86Box/src/include/86box/video.h:216:40: note: passing argument to parameter 'buf' here                                                                               extern void video_screenshot(uint32_t *buf, int start_x, int start_y, int row_len);
                                       ^
/data/data/com.termux/files/home/86Box/src/cdrom/cdrom_image_backend.c:656:17: error: expected expression
                [[fallthrough]];                                                                                   ^
1 error generated.
make[2]: *** [src/cdrom/CMakeFiles/cdrom.dir/build.make:90: src/cdrom/CMakeFiles/cdrom.dir/cdrom_image_backend.c.o] Error 1                                                                           make[1]: *** [CMakeFiles/Makefile2:600: src/cdrom/CMakeFiles/cdrom.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  2%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/fyl2x.cc.o
[  2%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_accumulate.c.o             [  3%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1531.c.o
[  4%] Building C object src/cpu/CMakeFiles/cpu.dir/fpu.c.o                                        1 warning generated.
[  4%] Building C object src/device/CMakeFiles/dev.dir/cassette.c.o
[  5%] Building C object src/unix/CMakeFiles/ui.dir/unix_cdrom.c.o
[  5%] Building C object src/device/CMakeFiles/dev.dir/cartridge.c.o
[  5%] Building C object src/cpu/CMakeFiles/cpu.dir/x86.c.o
[  5%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1541.c.o                            [  6%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_pentium.c.o
[  6%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat_poly.cc.o
[  6%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat.cc.o
[  6%] Built target ui                                                                             /data/data/com.termux/files/home/86Box/src/cpu/codegen_timing_pentium.c:823:42: warning: format specifies type 'unsigned long long' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]
        fatal("Illegal COUNT %016llx\n", timings);
                             ~~~~~~~     ^~~~~~~
                             %016lx
[  6%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat16.cc.o
[  6%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_p6.c.o
[  6%] Building C object src/cpu/CMakeFiles/cpu.dir/808x.c.o
[  7%] Building C object src/device/CMakeFiles/dev.dir/hasp.c.o
[  7%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1543.c.o
[  8%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat-muladd.cc.o
[  8%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_allocator.c.o
[  8%] Building C object src/device/CMakeFiles/dev.dir/hwm.c.o
1 warning generated.
[  9%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_block.c.o
[  9%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_winchip.c.o
[  9%] Building C object src/device/CMakeFiles/dev.dir/hwm_lm75.c.o
[  9%] Building C object src/device/CMakeFiles/dev.dir/hwm_lm78.c.o
[ 10%] Building C object src/device/CMakeFiles/dev.dir/hwm_gl518sm.c.o
[ 10%] Building C object src/cpu/CMakeFiles/cgt.dir/codegen_timing_winchip2.c.o
[ 10%] Building C object src/device/CMakeFiles/dev.dir/hwm_vt82c686.c.o
[ 10%] Building C object src/chipset/CMakeFiles/chipset.dir/ali1621.c.o
[ 10%] Building C object src/device/CMakeFiles/dev.dir/ibm_5161.c.o
[ 10%] Building C object src/chipset/CMakeFiles/chipset.dir/ali6117.c.o
[ 10%] Building C object src/device/CMakeFiles/dev.dir/isamem.c.o
/data/data/com.termux/files/home/86Box/src/chipset/ali1621.c:113:21: error: expected expression
                    [[fallthrough]];
                    ^
/data/data/com.termux/files/home/86Box/src/chipset/ali1621.c:128:21: error: expected expression
                    [[fallthrough]];
                    ^
2 errors generated.
make[2]: *** [src/chipset/CMakeFiles/chipset.dir/build.make:202: src/chipset/CMakeFiles/chipset.dir/ali1621.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 10%] Building C object src/device/CMakeFiles/dev.dir/isartc.c.o
[ 10%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat-round-pack.cc.o
/data/data/com.termux/files/home/86Box/src/chipset/ali6117.c:238:21: error: expected expression
                    [[fallthrough]];
                    ^
1 error generated.
make[2]: *** [src/chipset/CMakeFiles/chipset.dir/build.make:216: src/chipset/CMakeFiles/chipset.dir/ali6117.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:626: src/chipset/CMakeFiles/chipset.dir/all] Error 2
[ 10%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloat-specialize.cc.o
[ 10%] Built target cgt
[ 10%] Building CXX object src/cpu/softfloat/CMakeFiles/softfloat.dir/softfloatx80.cc.o
[ 10%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ir.c.o
[ 11%] Building C object src/device/CMakeFiles/dev.dir/__/lpt.c.o
[ 11%] Building C object src/device/CMakeFiles/dev.dir/pci_bridge.c.o
[ 11%] Building C object src/cpu/CMakeFiles/cpu.dir/386.c.o
[ 11%] Building C object src/cpu/CMakeFiles/cpu.dir/386_common.c.o
[ 11%] Building C object src/device/CMakeFiles/dev.dir/postcard.c.o
[ 11%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops.c.o
[ 11%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_3dnow.c.o
[ 11%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_branch.c.o
[ 11%] Building C object src/device/CMakeFiles/dev.dir/serial.c.o
[ 12%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_arith.c.o
[ 12%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_fpu_arith.c.o
[ 12%] Building C object src/device/CMakeFiles/dev.dir/clock_ics9xxx.c.o                           [ 13%] Building C object src/device/CMakeFiles/dev.dir/isapnp.c.o
[ 13%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_fpu_constant.c.o
[ 13%] Built target softfloat
[ 13%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_fpu_loadstore.c.o
[ 13%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_fpu_misc.c.o
[ 14%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_helpers.c.o
[ 14%] Building C object src/device/CMakeFiles/dev.dir/i2c.c.o
[ 14%] Building C object src/device/CMakeFiles/dev.dir/i2c_gpio.c.o
[ 14%] Building C object src/device/CMakeFiles/dev.dir/smbus_piix4.c.o
[ 14%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_jump.c.o
[ 14%] Building C object src/device/CMakeFiles/dev.dir/smbus_ali7101.c.o
[ 15%] Building C object src/device/CMakeFiles/dev.dir/keyboard.c.o                                /data/data/com.termux/files/home/86Box/src/device/smbus_piix4.c:200:25: error: expected expression
                        [[fallthrough]];
                        ^
/data/data/com.termux/files/home/86Box/src/device/smbus_piix4.c:221:25: error: expected expression
                        [[fallthrough]];
                        ^
/data/data/com.termux/files/home/86Box/src/device/smbus_piix4.c:256:25: error: expected expression
                        [[fallthrough]];
                        ^
3 errors generated.
make[2]: *** [src/device/CMakeFiles/dev.dir/build.make:356: src/device/CMakeFiles/dev.dir/smbus_piix4.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 15%] Building C object src/device/CMakeFiles/dev.dir/keyboard_xt.c.o
[ 15%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_logic.c.o
/data/data/com.termux/files/home/86Box/src/device/smbus_ali7101.c:198:21: error: expected expression
                    [[fallthrough]];
                    ^
1 error generated.
make[2]: *** [src/device/CMakeFiles/dev.dir/build.make:370: src/device/CMakeFiles/dev.dir/smbus_ali7101.c.o] Error 1
[ 15%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_misc.c.o
[ 15%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_arith.c.o
make[1]: *** [CMakeFiles/Makefile2:782: src/device/CMakeFiles/dev.dir/all] Error 2
[ 16%] Building C object src/cpu/CMakeFiles/cpu.dir/386_dynarec.c.o
[ 17%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_cmp.c.o
[ 17%] Building C object src/cpu/CMakeFiles/cpu.dir/x86_ops_mmx.c.o
[ 17%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_loadstore.c.o
[ 17%] Building C object src/cpu/CMakeFiles/cpu.dir/x86seg.c.o
[ 17%] Building C object src/cpu/CMakeFiles/cpu.dir/x87.c.o
[ 17%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_logic.c.o
[ 17%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_pack.c.o
[ 17%] Building C object src/cpu/CMakeFiles/cpu.dir/x87_timings.c.o
[ 18%] Building C object src/cpu/CMakeFiles/cpu.dir/8080.c.o
[ 18%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mmx_shift.c.o
[ 18%] Building C object src/cpu/CMakeFiles/cpu.dir/386_dynarec_ops.c.o
[ 19%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_mov.c.o
[ 19%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_shift.c.o
[ 19%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_ops_stack.c.o
[ 19%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_reg.c.o                    [ 20%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_backend_arm64.c.o
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64.c:329:16: warning: value size does not match register size specified by the constraint and modifier [-Wasm-operand-widths]
        : "=r"(cpu_state.old_fp_control));
               ^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64.c:328:14: note: use constraint modifier "w"
    asm("mrs %0, fpcr\n"
             ^~
             %w0
1 warning generated.
[ 20%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_backend_arm64_ops.c.o
[ 20%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_backend_arm64_uops.c.o
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_ops.c:343:38: warning: format specifies type 'unsigned long long' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]
        fatal("ADD_IMM_X %016llx\n", imm_data);
                         ~~~~~~~     ^~~~~~~~
                         %016lx                                                                    [ 20%] Building C object src/codegen_new/CMakeFiles/dynarec.dir/codegen_backend_arm64_imm.c.o
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_ops.c:665:34: warning: format specifies type 'unsigned int' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]
        fatal("CMNX_IMM %08x\n", imm_data);
                        ~~~~     ^~~~~~~~
                        %08lx
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_ops.c:686:34: warning: format specifies type 'unsigned int' but the argument has type 'uint64_t' (aka 'unsigned long') [-Wformat]
        fatal("CMPX_IMM %08x\n", imm_data);
                        ~~~~     ^~~~~~~~
                        %08lx
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:556:42: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("codegen_FABS %02x %02x %02x\n", uop->dest_reg_a_real, uop->src_reg_a_real);
                                      ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:569:42: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("codegen_FCHS %02x %02x %02x\n", uop->dest_reg_a_real, uop->src_reg_a_real);                                               ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:582:43: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("codegen_FSQRT %02x %02x %02x\n", uop->dest_reg_a_real, uop->src_reg_a_real);
                                       ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:1551:30: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("PF2ID %02x %02x\n", uop->dest_reg_a_real);
                          ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:1658:30: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("PFRCP %02x %02x\n", uop->dest_reg_a_real);
                          ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:1675:32: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("PFRSQRT %02x %02x\n", uop->dest_reg_a_real);                                                                    ~~~^
/data/data/com.termux/files/home/86Box/src/codegen_new/codegen_backend_arm64_uops.c:1701:30: warning: more '%' conversions than data arguments [-Wformat-insufficient-args]
        fatal("PI2FD %02x %02x\n", uop->dest_reg_a_real);                                                                    ~~~^
7 warnings generated.
3 warnings generated.
[ 20%] Built target dynarec
[ 20%] Built target cpu
make: *** [Makefile:156: all] Error 2
~/86Box $
```

Well, something's fucky:
```/data/data/com.termux/files/home/86Box/src/chipset/ali1621.c:113:21: error: expected expression
                    [[fallthrough]];
                    ^```
Meanwhile, line 113:
```#endif
```

--- a/src/include/86box/plat.h
+++ b/src/include/86box/plat.h
@@ -55,7 +55,7 @@ extern int strnicmp(const char *s1, const char *s2, size_t n);
 #    define off64_t  off_t
 #endif
 
-#if !defined (__APPLE__) || !defined(__clang__)
+#if !defined (__APPLE__) && !defined(__clang__)
 #    define FALLTHROUGH_ANNOTATION
 #endif
 

--- a/src/include/86box/plat_fallthrough.h
+++ b/src/include/86box/plat_fallthrough.h
@@ -18,7 +18,7 @@
 #ifndef EMU_PLAT_FALLTHROUGH_H
 #define EMU_PLAT_FALLTHROUGH_H
 
-#if !defined (__APPLE__) || !defined(__clang__)
+#if !defined (__APPLE__) && !defined(__clang__)
 #    define FALLTHROUGH_ANNOTATION
 #endif
 

